### PR TITLE
Added public announcements endpoint

### PR DESCRIPTION
**Pull request by Pchelolo**
Pending deployment of https://gerrit.wikimedia.org/r/#/c/320712/
Bug: https://phabricator.wikimedia.org/T150181
cc @wikimedia/services

### CODE REVIEW

**Comment by Pchelolo** [comment]
One Q is whether we want the endpoint to show up in the docs in all wikipedias or not. Habing it everywhere is easy and it's what this PR is doing. Limiting it to just enwiki is possible, but would require setting up a new project for it, a config deploy etc, so it's much more involved. Thoughts? Opinions?

**Comment by Pchelolo** [comment]
Tests are currently failing due to parsoid issue in beta cluster.

**Comment by gwicke** [comment]
> One Q is whether we want the endpoint to show up in the docs in all wikipedias or not. Habing it everywhere is easy and it's what this PR is doing. Limiting it to just enwiki is possible, but would require setting up a new project for it, a config deploy etc, so it's much more involved. Thoughts? Opinions?

If we returned empty lists of announcements for all languages, then exposing the end point as experimental everywhere could work. The API would work consistently everywhere, but wouldn't show any announcements for non-English languages. Of course, this assumes that the intention is to add support for non-English languages soon(ish).

@mdholloway @berndsi: Does this sound like a sensible direction to you? Are you planning to add support for projects other than enwiki in the foreseeable future?

**Comment by berndsi** [comment]
Good idea. Yes, the idea is to open this up to other wikis in the future.

**Comment by berndsi** [comment]
Or should we do an empty response (204), without payload?

**Comment by berndsi** [comment]
Proposal at https://gerrit.wikimedia.org/r/#/c/322016/

status: 200
`{ announce: [] }`

**Comment by berndsi** [comment]
> One Q is whether we want the endpoint to show up in the docs in all wikipedias or not.

Having it hidden would be slightly preferable right now. It's only meant to be consumed by apps, and of course experimental, at this time. But it's not a big deal if it shows up, too. But maybe you could add a note that this is not intended for third-party use.

File: v1/announcements.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      produces:
+        - application/json
+        - application/problem+json
+      responses:
+        '200':
+          description: Announcements for the given Wiki
+          schema:
+            $ref: '#/definitions/announcementsResponse'
+        '404':
+          description: Unsupported Wiki
```
Comment (berndsi) [inline v1/announcements.yaml]: Should be removed since I removed in in MCS, too.

File: v1/announcements.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        - title: Retrieve announcements
+          request:
+            params:
+                domain: en.wikipedia.org
+          response:
+            status: 200
+            headers:
+              content-type: application/json
+            body:
+              announce:
```
Comment (gwicke) [inline v1/announcements.yaml]: Lets remove the assertion in here that expects an announcement to be returned.

**Comment by Pchelolo** [comment]
This needs some updates

**Comment by Pchelolo** [comment]
@d00rman @berndsi Updated

File: v1/announcements.yaml [hunk 3]
```diff
@@ -35,28 +27,7 @@ paths:
...
-                  start_time: /.+/
-                  end_time: /.+/
-                  platforms: [ /.+/ ]
-                  text: /.+/
-                  action:
-                    title: /.+/
-                    url: /.+/
-                  caption_HTML: /.+/
-                  countries: [ /.+/ ]
+      x-monitor: false
```
Comment (gwicke) [inline v1/announcements.yaml]: Hmm, should we still have some basic monitoring? Just status 200 & the presence of an announce property would already ensure that we didn't break this end point in MCS or RB.

Comment (Pchelolo) [inline v1/announcements.yaml]: The checker script can't check just presence of the property unfortunately. But I can add a test for 200 status code

**Comment by gwicke** [comment]
LGTM overall.

**Comment by Pchelolo** [comment]
@gwicke Updated to fix your latest comments

**Comment by gwicke** [comment]
Waiting for travis..

**Pull request merged**